tools/rules_cgo.patch

```diff
@@ -0,0 +1,14 @@
+diff --git a/go/private/rules/cgo.bzl b/go/private/rules/cgo.bzl
+index b8fc93a6..e953d234 100644
+--- a/go/private/rules/cgo.bzl
++++ b/go/private/rules/cgo.bzl
+@@ -144,7 +144,9 @@ def cgo_configure(go, srcs, cdeps, cppopts, copts, cxxopts, clinkopts):
+                         # libclntsh.dylib.12.1, users have to create a unversioned symbolic link,
+                         # so it can be treated as a simple shared library too.
+                         continue
++                lib_opts.append("-Wl,--whole-archive")
+                 lib_opts.append(lib.path)
++                lib_opts.append("-Wl,--no-whole-archive")
```

Comment:
What do these flags do and why are they necessary?

Reply:
"--whole-archive" flag is set to make each object in the archive after this line to be included in the resulting binary object, which is runsc-plugin-stack in this case. Since we include TLDK and DPDK as plugin stack as an example, during initializing TLDK stack, it will need to use virtio_user driver defined in DPDK. If we do not set the object as whole archive, it will fail to find the corresponding driver.

Reply:
More comments have been added in this patch. Hope it may be helpful to make it more readable.